packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -152,4 +730,62 @@ export class SankeyChartBase extends React.Component<
       }
     });
   }
+  /**
+   *
+   * @param text is the text which we are trying to truncate
+   * @param rectangleWidth is the width of the rectangle which will contain the text
+   * @param padding is the space we need to leave between the rect lines and other text
+   * @param nodeWeight is the text if present needs to be accomodate in the same line as text.
+   * @returns the truncated text , if truncated given the above parameters.
+   */
+  private _truncateText(text: string, rectangleWidth: number, padding: number) {
+    const textLengthForNodeName = rectangleWidth - padding;
+    let elipsisLength = 0;
+    const tspan = select('.nodeName')
+      .append('text')
+      .attr('class', 'tempText')
+      .attr('font-size', '10')
+      .append('tspan')
+      .text(null);
+    tspan.text('...');
+    if (tspan.node() !== null) {
+      elipsisLength = tspan.node()!.getComputedTextLength();
```

Comment:
Add comments explaining the major logic points wherever applicable.

Reply:
Sure , i will add the comments.